libs/test/SalesforceSDKTest/src/com/salesforce/androidsdk/util/BundleTestHelper.kt

```diff
@@ -0,0 +1,49 @@
+package com.salesforce.androidsdk.util
+
+import android.os.Bundle
+import org.junit.Assert
+
+class BundleTestHelper {
+    companion object {
+        @JvmStatic
+        fun checkSameBundle(message: String, expected: Bundle?, actual: Bundle?) {
+            // Check if both bundles are null
+            if (expected == null && actual == null) {
+                return
+            }
+
+            // Fail if one is null and the other is not
+            if (expected == null || actual == null) {
+                Assert.fail("$message: One of the bundles is null.")
+                return
+            }
+
+            // Check if both bundles contain the same keys
+            for (key in expected.keySet()) {
```

Comment:
We could just compare the sizes and Assert.fail but then we would not know what fields are missing.